pkg/util/predicates.go

```diff
@@ -71,6 +73,50 @@ func (p FinalizersChangedPredicate) Update(e event.UpdateEvent) bool {
 		!reflect.DeepEqual(e.MetaOld.GetFinalizers(), e.MetaNew.GetFinalizers())
 }
 
+// FilterForOwner will only allow events that owned by noobaa
+type FilterForOwner struct {
+	OwnerSubstrings []string
+}
+
+// hasCorrectOwner checks if one of the owners has a substring that represents an expected owner
+func hasCorrectOwner(arr []v1.OwnerReference, ownerSubstrings []string) bool {
+	for _, b := range arr {
+		for _, sub := range ownerSubstrings {
+			if strings.Contains(b.Name, sub) {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// Create implements the create event trap for FilterForOwner
+func (p FilterForOwner) Create(e event.CreateEvent) bool {
+	eventOwners := e.Meta.GetOwnerReferences()
+	return hasCorrectOwner(eventOwners, p.OwnerSubstrings)
+
+}
+
+// Delete implements the delete event trap for FilterForOwner
+func (p FilterForOwner) Delete(e event.DeleteEvent) bool {
+	eventOwners := e.Meta.GetOwnerReferences()
+	return hasCorrectOwner(eventOwners, p.OwnerSubstrings)
+}
+
+// Update implements the update event trap for FilterForOwner
+func (p FilterForOwner) Update(e event.UpdateEvent) bool {
+	newEventOwners := e.MetaNew.GetOwnerReferences()
+	oldEventOwners := e.MetaOld.GetOwnerReferences()
+	return hasCorrectOwner(newEventOwners, p.OwnerSubstrings) || hasCorrectOwner(oldEventOwners, p.OwnerSubstrings)
```

Comment:
Why do you care about the old owner in this case? Why does it matter?

Reply:
right, they will be filtered out anyway by the ownerHandler, deleted.